### PR TITLE
Eulerian Tours editorial

### PR DESCRIPTION
**Pull request by brebenelmihnea**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 1]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+### Mail Delivery
+
+First let's define what an **Eulerian** path is
+
+> An Eulerian path is a path that goes through every edge once.
+
+Similarly, an **Eulerian cycle** is an Eulerian path that starts and ends
+with the same node.
+
+An important condition is that a graph can be an Eulerian graph if and only if every node has an even degree.
```
Comment (SansPapyrus683) [inline content/6_Advanced/Eulerian_Tours.mdx:33]: i thought a graph w/ 2 odd degree nodes also was possible?

Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:33]: Nope. It's necessary for an Eulerian path that all noded have even degree.
https://en.wikipedia.org/wiki/Eulerian_path

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 2]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+
+Now, to properly find the Eulerian cycle we run a slightly modified DFS.
+The DFS goes through only unvisited edges and the same edge can be processe multiple times throuoght the DFs, so we remove it from the graph at the first visit.
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:36]:
```suggestion
The DFS goes through only unvisited edges and the same edge can be processed multiple times throughout the DFS, so we remove it from the graph at the first visit.
```

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 3]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+
+The algorithm described is [Hierholzer's Algorithm](https://en.wikipedia.org/wiki/Eulerian_path).
+
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:39]:
```suggestion
## Implementation
```

add time complexity as well

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 4]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+	g.resize(n);
+	degree.resize(n);
+	seen.resize(m);
+
+	for (int i = 0; i < m; i++) {
+		int x, y;
+		cin >> x >> y;
+		x--, y--;
+		g[x].push_back({y, i});
+		g[y].push_back({x, i});
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:75]: emplace_back ? optional

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 5]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int n, m;
+vector<vector<pair<int, int>>> g;
+vector<int> degree, path;
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:49]: does degree need to be global?

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 6]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+
+	reverse(path.begin(), path.end());
+	if (path.size() != m + 1 || path.back() != n) {
+		cout << "IMPOSSIBLE";
+	} else {
+		for (auto node : path) { cout << node << ' '; }
+	}
 
-<IncompleteSection />
+	return 0;
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:156]:
```suggestion
```

File: content/6_Advanced/Eulerian_Tours.mdx [hunk 7]
```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
...
+	}
+}
+```
+
+</CPPSection>
+</LanguageSection>
+
+### Teleporters
+
+The condition of existance for an eulerian path in a directed graph is: At most one node has $out_i - in_i=1$ and at most one node has $in_i - out_i=1$.
```
Comment (ryanchou-dev) [inline content/6_Advanced/Eulerian_Tours.mdx:102]: can u explain why?

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 2 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:36]: Alright

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 3 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:39]: Ok

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 6 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:156]: Done

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 4 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:77]: Ok changed

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 5 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:51]: OK, it's local now

File: content/6_Advanced/Eulerian_Tours.mdx (hunk 7 above)
Comment (brebenelmihnea) [reply content/6_Advanced/Eulerian_Tours.mdx:104]: Ok  I'll explain

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**